internal/pkg/agent/application/gateway/fleet/fleet_gateway.go

```diff
@@ -318,8 +334,10 @@ func (f *FleetGateway) convertToCheckinComponents(components []runtime.Component
 	// OTel status is placed as a component for each top-level component in OTel
 	// and each subcomponent is a unit.
 	if collector != nil {
-		for id, item := range collector.ComponentStatusMap {
-			state, msg := otelhelpers.StateWithMessage(item)
+		sortedIds := slices.Sorted(maps.Keys(collector.ComponentStatusMap))
```

Comment:
why do we need to sort the keys of the `collector.ComponentStatusMap` ? Especially since we then re-access the map by key ?
Same questions for the unites below)

Reply:
The function takes a map as an input and returns a slice. Having the output order be consistent is a good practice in general, but here in particular I did it to make testing easier. testify has better diffs for ordered slices.

Reply:
If it's only for test assertions there's https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch that would still work without requiring sorting.
Having the output order consistent may not necessarily be a good practice, if it's not guaranteed by contract but consumer come to rely on the order (moreover if the output get marshaled/unmarshaled as an object order is not guaranteed once more).